Il2CppInterop.Generator/Passes/Pass61ImplementAwaiters.cs

```diff
@@ -0,0 +1,119 @@
+                if (interfaceImplementation is null)
+                    continue;
+
+                var allOnCompleted = typeContext.NewType.Methods.Where(m => m.Name == nameof(INotifyCompletion.OnCompleted)).ToArray();
+                if (allOnCompleted.Length == 0)
+                {
+                    // Likely defined as INotifyCompletion.OnCompleted & the name is unhollowed as something like "System_Runtime_CompilerServices_INotifyCompletion_OnCompleted"
+                    allOnCompleted = typeContext.NewType.Methods.Where(m => ((string?)m.Name)?.EndsWith(nameof(INotifyCompletion.OnCompleted)) ?? false).ToArray();
+                    var typeName = typeContext.OriginalType.FullName;
+                    Logger.Instance.LogInformation("Found explicit implementation of INotifyCompletion on {typeName}", typeName);
+                }
+
+                // Conversion spits out an Il2CppSystem.Action, so look for methods that take that (and only that) in & return void, so the stack is balanced
+                // And use IsAssignableTo because otherwise equality checks would fail due to the TypeSignatures being different references
+                var interopOnCompleted = allOnCompleted.FirstOrDefault(m => m.Parameters.Count == 1 && m.Signature is not null && m.Signature.ReturnType == voidRef && SignatureComparer.Default.Equals(m.Signature.ParameterTypes[0], actionConversion.Signature?.ReturnType));
+
+                if (interopOnCompleted is null)
+                {
+                    var typeName = typeContext.OriginalType.FullName;
+                    var foundMethodCount = allOnCompleted.Length;
+                    Logger.Instance.LogInformation("Type {typeName} was found to implement INotifyCompletion, but no suitable method was found. {foundMethodCount} method(s) were found with the required name.", typeName, foundMethodCount);
+                    continue;
+                }
+
+                var cloner = new MemberCloner(typeContext.NewType.Module, new ParameterCloneListener(actionUntypedRef.Value.ToTypeSignature()))
+                    .Include(interopOnCompleted);
+                var cloneResult = cloner.Clone();
+
+                // Established that INotifyCompletion.OnCompleted is implemented, & interop method is defined, now clone it to create the .NET interface implementation method that jumps straight to it
+                var proxyOnCompleted = (MethodDefinition)cloneResult.ClonedMembers.Single();
+                proxyOnCompleted.Signature!.ParameterTypes[0] = actionUntypedRef.Value.ToTypeSignature();
+                var parameter = proxyOnCompleted.Parameters[0].GetOrCreateDefinition();
+
+                var body = proxyOnCompleted.CilMethodBody ??= new(proxyOnCompleted);
+
+                typeContext.NewType.Interfaces.Add(new(notifyCompletionRef.Value));
+
+                var instructions = body.Instructions;
+                instructions.Add(CilOpCodes.Nop);
```

Comment:
This is unnecessary.

Reply:
The nops are there because I was copying the IL I got from a normally-compiled OnComplete. I'll remove them.